packages/contracts/test/src/plume/AggregateTokenTest.t.sol

```diff
@@ -0,0 +1,132 @@
+            _liquidVault.balanceOf(address(aggregateToken), _liquidVault.currentPeriod()),
+            "AggregateToken should receive ERC1155 token shares equal to the depositAmount"
+        );
+        assertEq(0, _asset.balanceOf(address(aggregateToken)), "There shouldn't be any remaining amount");
+    }
+
+    function test__AggregateTokenTest__SellComponentToken() public {
+        TestParamSet.TestParam memory testParams =
+            TestParamSet.TestParam({ principal: 2_000 * _scale, depositPeriod: 10, redeemPeriod: 70 });
+
+        // Move the blocktime to depositPeriod
+        _warpToPeriod(_liquidVault, testParams.depositPeriod);
+
+        _depositAssets_AggregateToken(alice, testParams.principal);
+
+        assertEq(
+            testParams.principal,
+            _asset.balanceOf(address(aggregateToken)),
+            "The asset balance of AggregateToken should be equal to the principal, as it is the first deposit"
+        );
+
+        // Call buyComponentToken
+        vm.startPrank(NEST_ADMIN_ADDRESS);
+        aggregateToken.approveComponentToken(IComponentToken(address(_liquidVault)), testParams.principal);
+        aggregateToken.buyComponentToken(IComponentToken(address(_liquidVault)), testParams.principal);
+        vm.stopPrank();
+
+        assertEq(0, _asset.balanceOf(address(aggregateToken)), "There shouldn't be any remaining amount");
+        assertEq(
+            testParams.principal,
+            _liquidVault.balanceOf(address(aggregateToken), testParams.depositPeriod),
+            "AggregateToken should receive ERC1155 token shares equal to the principal"
+        );
+
+        // Invest assets to the liquidStone
+        uint256 investAmount = 2_000 * _scale;
+        _investToLiquidStone(investAmount);
+
+        // Move the blocktime to request redeem
+        _warpToPeriod(_liquidVault, testParams.redeemPeriod - _liquidVault.noticePeriod());
+
+        // Call requestSellComponentToken (newly added by us)
+        vm.prank(NEST_ADMIN_ADDRESS);
+        aggregateToken.requestSellComponentToken(IComponentToken(address(_liquidVault)), testParams.principal);
+
+        // Move the blocktime to redeemPeriod
+        _warpToPeriod(_liquidVault, testParams.redeemPeriod);
+
+        // Call sellComponentToken
+        vm.prank(NEST_ADMIN_ADDRESS);
+        aggregateToken.sellComponentToken(IComponentToken(address(_liquidVault)), testParams.principal);
+
+        uint256 expectedAmount = testParams.principal + _expectedReturns(0, _liquidVault, testParams);
+        assertEq(expectedAmount, _asset.balanceOf(address(aggregateToken)));
+        assertEq(0, _liquidVault.balanceOf(address(aggregateToken), testParams.depositPeriod));
+        assertEq(investAmount + testParams.principal - expectedAmount, _asset.balanceOf(address(_liquidVault)));
+    }
+
+    /// @dev Deposits assets into AggregateToken
+    function _depositAssets_AggregateToken(address user, uint256 depositAmount) internal {
```

Comment:
please rename to be _depositAssetsAggregateToken

we shouldn't use both snake case and camel case.